### PR TITLE
ddl: check no partitions error when create partition table

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 

In MySQL
```SQL
mysql > create table t (a int not null) partition by range( a );
(1492, u'For RANGE partitions each partition must be defined')
mysql > create table t (a int not null) partition by range columns ( a );
(1492, u'For RANGE partitions each partition must be defined')
mysql > create table th (a int, b int) partition by hash(a) partitions 0;
(1504, u'Number of partitions = 0 is not an allowed value')
```
But in TiDB will execute those sql successful. then add index in this no partition table will panic.

Other check, drop partition will have check if drop the last partition and return `ErrDropLastPartition`, so drop partition will not cause the no partition table problem.

### What is changed and how it works?
Add check when create partition table.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by crazycs520** [comment]
@tiancaiamao @winkyao @zimulala PTAL

**Comment by tiancaiamao** [comment]
LGTM

File: ddl/db_partition_test.go [hunk 1]
```diff
@@ -233,6 +233,13 @@ func (s *testIntegrationSuite) TestCreateTableWithPartition(c *C) {
 		  partition by range columns (a, b)
 		  (partition p0 values less than (10, 10.0))`)
 	c.Assert(ddl.ErrUnsupportedPartitionByRangeColumns.Equal(err), IsTrue)
+
+	_, err = tk.Exec(`create table t31 (a int not null) partition by range( a );`)
+	c.Assert(ddl.ErrPartitionsMustBeDefined.Equal(err), IsTrue)
+	_, err = tk.Exec(`create table t32 (a int not null) partition by range columns( a );`)
+	c.Assert(ddl.ErrPartitionsMustBeDefined.Equal(err), IsTrue)
+	_, err = tk.Exec(`create table t33 (a int, b int) partition by hash(a) partitions 0;`)
+	c.Assert(ddl.ErrNoParts.Equal(err), IsTrue)
```
Comment (zimulala) [inline ddl/db_partition_test.go]: Could we use `assertErrorCode`？

Comment (crazycs520) [reply ddl/db_partition_test.go]: done.

**Pull request merged**